### PR TITLE
journalist: init at 1.0.0-unstable-2024-05-28

### LINKED ISSUE #317832: Package request: journalist
**Project description**
Journalist is an RSS aggregator that can sync subscriptions and read/unread items across multiple clients without requiring a special client-side integration. Clients can use Journalist by simply subscribing to its personalized RSS feed.

**Metadata**

* homepage URL: https://github.com/mrusme/journalist/tree/master
* source URL: https://github.com/mrusme/journalist/tree/master
* license: gpl3
* platforms: linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Moraxyc**
## Description of changes

Closes #317832

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by hakan-demirli** [comment]
Tested this one with https://github.com/NixOS/nixpkgs/pull/318317 since it depends on this one. I haven't encountered any problems.

File: pkgs/by-name/jo/journalist/go.sum
Comment (0x61nas) [inline pkgs/by-name/jo/journalist/go.sum:1]: is this necessary? I see the upstream has one already

File: pkgs/by-name/jo/journalist/package.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    "-w"
+    "-X github.com/mrusme/journalist/journalistd.VERSION=${version}"
+  ];
+
+  meta = with lib; {
+    description = "An RSS aggregator";
+    homepage = "https://github.com/mrusme/journalist";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ moraxyc ];
+    mainProgram = "journalist";
```
Comment (0x61nas) [inline pkgs/by-name/jo/journalist/package.nix:37]:
```suggestion
    mainProgram = "journalist";
    platforms = with lib.platforms; unix ++ windows;
    broken = stdenv.isDarwin; # needing some apple_sdk packages
```

File: pkgs/by-name/jo/journalist/go.sum
Comment (Moraxyc) [reply pkgs/by-name/jo/journalist/go.sum:1]: There are some mistakes in the upstream go.sum file that this file fixes.

```
       >        github.com/swaggo/swag: missing go.sum entry for module providing package github.com/swaggo/swag (imported by github.com/mrusme/journalist/docs); to add:
```

File: pkgs/by-name/jo/journalist/package.nix (hunk 1 above)
Comment (Moraxyc) [reply pkgs/by-name/jo/journalist/package.nix:37]: Thanks for testing it on Darwin!

Comment (0x61nas) [reply pkgs/by-name/jo/journalist/package.nix:38]: i did not test it, but mark it as a broken until someone tests it. seams pretty reasonable default to me.

Comment (0x61nas) [reply pkgs/by-name/jo/journalist/package.nix:38]: and yeah, you have to import `stdenv` for this to work, i forget to include it in the suggestion, sorry

Comment (Moraxyc) [reply pkgs/by-name/jo/journalist/package.nix:38]: I added it on top.

Language server is pretty good in this case😄

File: pkgs/by-name/jo/journalist/go.sum
Comment (0x61nas) [reply pkgs/by-name/jo/journalist/go.sum:1]: if it is just one or two lines, can't be done via patch?

Comment (Moraxyc) [reply pkgs/by-name/jo/journalist/go.sum:1]: Let me check it

Comment (Moraxyc) [reply pkgs/by-name/jo/journalist/go.sum:1]: `go get` almost changed everything in `go.sum`, I have no idea if this type of fix up is standard for go package.

Comment (0x61nas) [reply pkgs/by-name/jo/journalist/go.sum:1]: oh, in this case, i think it's fine to hold our own `go.sum`, until the upstream get fixed

**Review by 0x61nas — APPROVED** [review]

File: pkgs/by-name/jo/journalist/package.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  ];
+
+  meta = {
+    description = "RSS aggregator";
+    homepage = "https://github.com/mrusme/journalist";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ moraxyc ];
+    mainProgram = "journalist";
+    platforms = with lib.platforms; unix ++ windows;
+    broken = stdenv.isDarwin; # needing some apple_sdk packages
```
Comment (SuperSandro2000) [inline pkgs/by-name/jo/journalist/package.nix:40]: Why not just add them?

File: pkgs/by-name/jo/journalist/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    "-X github.com/mrusme/journalist/journalistd.VERSION=${version}"
+  ];
+
+  meta = {
+    description = "RSS aggregator";
+    homepage = "https://github.com/mrusme/journalist";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ moraxyc ];
+    mainProgram = "journalist";
+    platforms = with lib.platforms; unix ++ windows;
```
Comment (SuperSandro2000) [inline pkgs/by-name/jo/journalist/package.nix:39]: I don't think we support go windows even

File: pkgs/by-name/jo/journalist/package.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    repo = "journalist";
+    rev = "eda6ce67c975f928ba2dfe94a39b69d3f864d051";
+    hash = "sha256-pxWBvh7j+IRvGLpQA0TEqorw7f0OZtXux5svMMqVooA=";
+  };
+
+  overrideModAttrs = _oldAttrs: {
+    postPatch = ''
+      cp ${./go.sum} go.sum
+    '';
+  };
```
Comment (SuperSandro2000) [inline pkgs/by-name/jo/journalist/package.nix:23]:
```suggestion
```
I am 50% sure that patches are imported into the vendor automatically

File: pkgs/by-name/jo/journalist/go.sum
Comment (SuperSandro2000) [reply pkgs/by-name/jo/journalist/go.sum:1]: Please send a PR to upstream to fix their lock file and fetch that instead.

File: pkgs/by-name/jo/journalist/package.nix (hunk 2 above)
Comment (Moraxyc) [reply pkgs/by-name/jo/journalist/package.nix:40]: Ah…I don’t have any darwin device, this line is actually suggested by @0x61nas .
https://github.com/NixOS/nixpkgs/pull/318324#discussion_r1632130955

**Pull request merged**